File: BinarySearchTree/BinarySearchTree.java
package BinarySearchTree;

public class BinarySearchTree {
    Node root;
    public void insert(Node node){
        root = insertHelper(root, node);
    }

    private Node insertHelper(Node root, Node node){
        int data = node.data;

        if(root == null){
            root = node;
            return root;
        }
        else if(root.data < data){
            root.right = insertHelper(root.right, node);
        }
        else{
            root.left = insertHelper(root.left, node);
        }
        return root;
    }

    public void display(){
        displayHelper(root);
    }
    private void displayHelper(Node root){
        if(root != null){
            displayHelper(root.left);
            System.out.print(root.data + " ");
            displayHelper(root.right);
        }
    }

    public boolean search(int  data){
        return searchHelper(root, data);
    }

    private boolean searchHelper(Node root , int data){
        if(root == null){
            return false;
        }
        else if (root.data == data){
            return true;
        }
        else if (root.data > data){
            return searchHelper(root.left, data);
        }
        else {
            return searchHelper(root.right, data);
        }
    }

    public void remove(){}

    public void removeHelper(){}


}

File: c-sharp-dsa-implementation-codes/Data-Structures-and-Algorithms/Binary-search-tree/BinarySearchTree.cs
using Data_Structures_and_Algorithms.Graph_Adjecancy_Matrix;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection.Metadata.Ecma335;
using System.Text;
using System.Threading.Tasks;

namespace Data_Structures_and_Algorithms.Binary_search_tree
{
    class BinarySearchTree
    {
        BSTNode root;

        public void Insert(BSTNode newNode)
        {
            root = InsertHelper(root, newNode);
        }

        public BSTNode InsertHelper(BSTNode root, BSTNode newNode)
        {
            int data = newNode.data;

            if (root == null)
            {
                root = newNode;
                return root;
            }
            else if (root.data < newNode.data)
            {
                root.right = InsertHelper(root.right, newNode);
            }
            else 
            {
                root.left = InsertHelper(root.left, newNode);
            }

            return root;
        }

        public void Display() 
        {
            DisplayHelper(root);
        }

        public void DisplayHelper(BSTNode root)
        {
            if (root != null)
            {
                DisplayHelper(root.left);
                Console.Write(root.data + ", ");
                DisplayHelper(root.right);
            }
        }

        public void DisplayInDescendingOrder()
        {
            DisplayDescendingOrderHelper(root);
        }

        public void DisplayDescendingOrderHelper(BSTNode root)
        {
            if (root != null)
            {
                DisplayDescendingOrderHelper(root.right);
                Console.Write(root.data + ", ");
                DisplayDescendingOrderHelper(root.left);
            }

        }

        public string Search(int data)
        {
            if (SearchHelper(root, data))
            {
                return "Your value found value is : " + data;
            }
            else 
            {
                return "You value has not been found, please try again";
            }
        }

        public Boolean SearchHelper(BSTNode root, int data)
        {
            if (root == null)
            {
                return false;
            }
            else if (root.data == data)
            {
                return true;
            }
            else if (root.data > data)
            {
                return SearchHelper(root.left, data);
            }
            else 
            {
                return SearchHelper(root.right, data);
            }
        }
    }
}
